### ISSUE TITLE
Individual compensations  X,Y,Z

### ISSUE THREAD

**Issue by boromyr** [SoftFever/OrcaSlicer#6562]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

No

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

It would be more useful and versatile to be able to compensate the individual axis instead of the XY shrinkage, adding a scale multiplier for each axis. This would simplify the calibration processes for each printer without modifying the firmware or steps per mm, or alternatively it can also be used to compensate for shrinkage.

### Describe alternatives you've considered

All calibration processes

**Comment by idkjustgoogleit** [comment]
+1

at least some form of z-height shrinkage compensation instead playing with top shell thickness settings, or scaling each print manually...

**Comment by Lab1717** [comment]
+1

**Comment by boromyr** [comment]
In the dev version I saw the compensation of the Z axis has been introduced, it should be quite simple therefore separating xy competition

**Comment by Sidiusz** [comment]
That's a good suggestion

**Comment by boromyr** [comment]
Up

**Comment by DusekM98** [comment]
I agree. I don't see a reason why it shouldn't be there. At least in the advanced options. Compensation is just a scale operation, so splitting it into X and Y can't be that difficult. It would also improve the accuracy due to shrinkage differences in X and Y.

**Comment by cclix** [comment]
+1

Would be useful on the Kobra 3, because as for now you can't set the Rotation Distance in the printer.cfg cause the firmware is locked up. Would be possible with custom firmware but not with stock. 🙄

**Comment by ianalexis** [comment]
Why X will have a different coeficient than Y?
If your model has some specific geometry and you cannot correctly compensate it with X/Y and Z shrinkage or [X-Y Compensation](https://github.com/SoftFever/OrcaSlicer/wiki/quality_settings_precision#x-y-compensation) you always can "compensate" that with the Scale tool.

<img width="429" height="216" alt="Image" src="https://github.com/user-attachments/assets/7deb6ed3-db81-4d10-8448-6b84173af6e2" />

> Rotation Distance

I think that you are refering to the E-steps? This is not what this Issue is talking about.

> at least some form of z-height shrinkage compensation instead playing with top shell thickness settings, or scaling each print manually...

this was done here #6507

**Comment by boromyr** [comment]
You need to manually set the scale for each model if you want to apply different retractions or dimensional corrections to compensate for small defects of your printer. Having them independent at the slicer level would mean greater freedom for anyone: you can use them to compensate for material shrinkage, or even to calibrate the step/mm without modifying firmware/.cfg not only at the machine level but also for each filament type.  
No printer supports modifying the step/mm per filament type, only globally; this way you can dimensionally calibrate your printer for any spool you have, or perhaps for other uses we haven't imagined yet.

**Comment by ianalexis** [comment]
Do you have any example parts where the measurement differs between X and Y?
I can't find any examples of cases where this is a real problem.
Based on what has been said [here](https://github.com/SoftFever/OrcaSlicer/issues/10695#issuecomment-3299268168), the differences are completely negligible and attributable to many other factors.

I am one of the main advocates of adding as much flexibility as possible for the user, but in this case, creating another point on the part scale (on top of material shrinkage, part scaling, and XY compensation) does not seem helpful and may cause more confusion.

**Comment by boromyr** [comment]
With separable variables, you can calibrate the steps/mm more easily; in fact, there's no need to calibrate the steps or the shrinkage anymore, you can calibrate both in a single step. The common offset of X and Y from 100% will correspond to the shrinkage, the discrepancy between the two will be the steps compensation. Then, anyone who wants can always calibrate both things in the traditional way.

**Comment by ianalexis** [comment]
> calibrate the steps/mm more easily

The slicer is not where you do that.

> X and Y from 100% will correspond to the shrinkage

Shrinkage is a material property, not a machine one.
How much the material shrink depends on the material/colorant/enfortecer mix, not a bad steps calibrated printer.

I think that you are looking for an "% Compensate printer motors" or something like that, in the Machine menu.

But that should be done in Machine G-code with:

- [Marlin](https://marlinfw.org/docs/gcode/M092.html)
- [Klipper](https://www.klipper3d.org/Rotation_Distance.html)


Im still waiting to see any real life test of this.

**Comment by boromyr** [comment]
It doesn't matter what you call it, the effect is the same; it's up to the user to decide how to use it.  
No proof is needed; it's mathematically elementary: if you scale a model by 200% or compensate the steps for mm to print 200% of the normal size, in both cases you'll get a model twice the original size. However, one thing that the step compensation in X or Y doesn't account for is that the extruder doesn't see the compensation of the XY motors, resulting in over-extrusion or under-extrusion. Consequently, a model printed with motors compensated at 200% will be under-extruded by up to 50% on both axes. A mass of filament creates a ghostly shape. And the motors are stressed twice as much.  
This is an exaggerated case to better explain the concept I want to convey.  

Therefore, the XYZ scaling in the slicer is better because:  

- You can use it to compensate for XY and Z shrinkage  
- You can use it to compensate for X, Y, and Z steps  
- The compensation does not affect extrusion because the slicer performs the correct calculations on the new dimensions  
- It does not require firmware modification, which is better for everyone, especially for those who do not have access or are less experienced  
- You can always choose not to use any of the above and stick with the current mode

**Comment by ianalexis** [comment]
> It doesn't matter what you call it,

It does matter and i think is one of the main problems here. Trying to use something to another thing that is not mean to.

> one thing that the step compensation in X or Y doesn't account

What do you mean with X or Y compensation? The shrinkage?

> * does not require firmware modification, which is better for everyone

its not, your printer is Slicer agnostic. Fix core printer errors with the slicer is not the best practice becouse after any update or using any other slicer it will break.

As i said earlier, you could use the Klipper or marlin commands to change whats wrong.


---

Really important thing:

- **Material** shrinkage is a **material** value.
- **Machine** steps is a **machine** value.
- **Model** size is a **model** value.

50x50x50

<img width="348" height="446" alt="Image" src="https://github.com/user-attachments/assets/975a3ddd-a059-431d-bfc2-eb1698196bed" />

Scaled 200%

<img width="551" height="796" alt="Image" src="https://github.com/user-attachments/assets/93637e63-510d-4e51-b0ea-8414b3ad7a66" />

material shrinkage 200%

<img width="236" height="324" alt="Image" src="https://github.com/user-attachments/assets/bd6f7c33-2ba4-4fde-b280-a606922bd181" />

-2mm contour compensation

<img width="1368" height="912" alt="Image" src="https://github.com/user-attachments/assets/51d6401b-6d0f-4e07-9a06-7440546e5905" />

**Comment by boromyr** [comment]
> It does matter and i think is one of the main problems here. 

How to name this feature/improvement is up to the developers; I am only presenting the concept.

> Trying to use something to another thing that is not mean to.

You can use Z compensation with the current version of Orca to also compensate dimensionally for your Z steps. It's an unconventional use.
 
> What do you mean with X or Y compensation? The shrinkage?

Sorry, maybe a lot was lost during the translation. I mean that if you adjust the steps/distance of rotation to get motor movements double the necessary, you will get a 200% scaled print. This modification is only for the X and Y movement (for the example we are currently discussing), but not for E; the extruder has its steps calibrated and does not account for how you, as the user, have modified the steps of the other motors, and consequently you will get under-extrusion. Basically, E does not know what happens to the other axes; it only extrudes the amount calculated by the slicer. If the motor movements have been modified, extrusion will inevitably be affected. If you extrude 10mm over a 50mm movement and then over a 60mm one, you will notice a significant difference. The compensation in the slicer solves both material retraction and motor steps in a single step.

> 
> its not, your printer is Slicer agnostic. Fix core printer errors with the slicer is not the best practice becouse after any update or using any other slicer it will break.

The user who uses other slicers or does not want this feature can always choose not to use it, precisely because of its switchability.

> Really important thing:
> 
> * **Material** shrinkage is a **material** value.
> * **Machine** steps is a **machine** value.
> * **Model** size is a **model** value.

And on this we agree, it is not debatable.

Scaling by 50% is like withdrawing 200%.
